### PR TITLE
pipewire.md: add a page on setting up pipewire

### PR DESCRIPTION
**Pull request by kkga**
Before submitting a pull request, please read [CONTRIBUTING](./CONTRIBUTING.md); pull requests that do not meet the criteria described there will not be merged. Note that this repository's CONTRIBUTING contains information specific to this repository, and is not the same as CONTRIBUTING for the void-packages repository.

We prioritise pull requests involving information specific to Void over those involving information applicable to Linux in general.

### CODE REVIEW

**Comment by st3r4g** [comment]
Looks good, can you add the Jack replacement info too from [my gist](https://gist.github.com/st3r4g/6c681a28b0403b3b02636f510ff68039)? I'll delete the gist after this is merged.

**Comment by kkga** [comment]
@st3r4g will do! Thanks for the gist :+1: This is mostly a copy-paste from it :D

**Comment by kkga** [comment]
Added a small section on JACK replacement too.

**Comment by st3r4g** [comment]
hm, it might be a bit too soon to "advertise" pipewire as viable to regular users. I'm just trying 0.3.19 and they changed the syntax for `exec` in `pipewire.conf`, so that if you modified the old config (most have done so to enable pulse), xbps keeps the old config and pw breaks.

File: src/config/media/pipewire.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+$ pactl info
+
+...
+Server Name: PulseAudio (on PipeWire 0.3.18)
+...
+```
+
+## ALSA integration
+
+Install `alsa-pipewire`, then enable the pipewire alsa device and make it
```
Comment (ericonr) [inline src/config/media/pipewire.md:37]:
```suggestion
Install `alsa-pipewire`, then enable the PipeWire ALSA device and make it
```

**Review by ericonr — COMMENTED** [review]
This is interesting, but I'm not totally sure all of it qualifies as Void specific info; the details about starting `pipewire` are important because we don't have systemd handling this for us, but the configuration part could (and probably should) be covered by upstream guides (which we could link to).

I think you also need to mention that pipewire requires a user dbus session? Unless I'm mistaken about that requirement.

**Comment by travankor** [comment]
Pipewire is also needed for screensharing, right?

**Comment by st3r4g** [comment]
> Pipewire is also needed for screensharing, right?

On Wayland, yes. Probably it should be mentioned in the Wayland page

**Comment by kkga** [comment]
Having issues getting bluetooth audio to work with `pipewire-pulse`.

If anyone managed to make pipewire work with bluetooth headsets, plz let me know so I can also update this PR accordingly.

Posted some observations here on the original gist by @st3r4g: https://gist.github.com/st3r4g/6c681a28b0403b3b02636f510ff68039#gistcomment-3604832

**Comment by bnhrkn** [comment]
> If anyone managed to make pipewire work with bluetooth headsets

@kkga I've been daily driving pipewire with bluetooth headset/headphones for several weeks now, works fine for me.

**Comment by kkga** [comment]
@bnhrkn that's great!
I haven't tested bluetooth for a while, but otherwise pipewire has been working great for me. Only downside is that the config format changes frequently, so I often have to manually change files in /etc/pipewire after a system upgrade.
Did you have to do anything specific for bluetooth to work?

**Comment by bnhrkn** [comment]
@kkga On top of the basic pirewire install steps, I had to get `bluez` and enable its `bluetoothd` service. I use the `blueman` GUI which makes connections straightforward. I also have `libspa-bluetooth`. Pulseaudio support in Pipewire must be correctly enabled so that Pipewire's bluez support works. Pavucontrol helps controls everything. If you want I can pastebin my pipewire configs so you can diff them with yours. Sorry if its not very detailed but it's been a while since I set this up. 
Both @st3r4g's gist, pipewire and bluetooth archwiki were helpful resources.

File: src/config/media/pipewire.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+desktop environment. If using [startx](../graphical-session/xorg.md#startx),
+add the following line in your `.xinitrc` file:
+
+```
+pipewire &
+```
+
+## PulseAudio replacement
+
+PipeWire can start a PulseAudio server. To enable it, uncomment the following
```
Comment (st3r4g) [inline src/config/media/pipewire.md:15]: I would now suggest to launch `pipewire-pulse` in the same place where you launch `pipewire`, instead of from the config. Keeping the config unmodified allows xbps to automatically replace it with the updated versions.

File: src/config/media/pipewire.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+# ln -s /usr/share/alsa/alsa.conf.d/99-pipewire-default.conf /etc/alsa/conf.d
+```
+
+## JACK replacement
+
+Install `libjack-pipewire` and use `pw-jack` to launch JACK clients:
+
+```
+$ pw-jack <application>
+```
```
Comment (st3r4g) [inline src/config/media/pipewire.md:52]: here you could mention how to set up the automatic libjack override via `/etc/ld.so.conf.d` (glibc) or `/etc/ld-musl-$ARCH.path` (musl)

File: src/config/media/index.md [hunk 4]
```diff
@@ -6,3 +6,6 @@ To setup audio on your Void Linux system you have to decide if you want to use
 [PulseAudio](./pulseaudio.md) or just [ALSA](./alsa.md).
 
 Some applications require PulseAudio, especially closed source programs.
```
Comment (paper42) [inline src/config/media/index.md:8]: You could mention PipeWire's pulse server.

**Comment by paper42** [comment]
bluetooth.md contains this line:

> To use an audio device such as a wireless speaker or headset, ALSA users need to install the bluez-alsa package. PulseAudio users do not need any additional software.

I think PipeWire should be mentioned here too. Users may want to set up bluetooth after they installed their audio system, so they might miss the pipewire install message.

File: src/config/media/pipewire.md (hunk 2 above)
Comment (kkga) [reply src/config/media/pipewire.md]: That's a great suggestion.
I've tried this on Sway, by executing both `pipewire` and `pipewire-pulse` and it works fine.

File: src/config/media/pipewire.md (hunk 3 above)
Comment (kkga) [reply src/config/media/pipewire.md:73]: I'm not sure what exactly this suggestion should be -- never worked with JACK.
I can see some notes on this in your gist, but I'm not sure what the actual steps should be. Can you give an example?

Comment (bnhrkn) [reply src/config/media/pipewire.md:73]: @kkga I these steps are available [on Reddit](https://www.reddit.com/r/voidlinux/comments/mjzlry/replacing_jack_with_pipewire/gteamw8?utm_source=share&utm_medium=web2x&context=3).

For glibc:
1. Create `/etc/ld.so.conf.d/pipewire-jack.conf` with content `/usr/lib/pipewire-0.3/jack`
2. `sudo ldconfig`

Worked for me.

Comment (kkga) [reply src/config/media/pipewire.md:75]: Added a note for glibc.
Can you post the exact steps needed for musl? Don't want to mess this up, since I have no way of testing.

Comment (st3r4g) [reply src/config/media/pipewire.md:80]: I have just tried it for musl and it's too easy to fuck up, so I don't feel like recommending it. Basically musl will not fall back to `/lib` unless explicitly told so, so creating that file is dangerous.

**Comment by thegarlynch** [comment]
Is there any tips for GDM ?

**Comment by kkga** [comment]
> Is there any tips for GDM ?

If you're asking about a way to autostart pipewire with GDM/Gnome: according to ArchWiki, you can use gnome-tweaks to do this https://wiki.archlinux.org/title/GNOME#Autostart

File: src/config/media/pipewire.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+PipeWire should be started as a user. Use the autostarting mechanism of your
+desktop environment.
+
+If using [startx](../graphical-session/xorg.md#startx), add the following line
+in your `.xinitrc` file:
+
+```
+pipewire &
+```
```
Comment (ericonr) [inline src/config/media/pipewire.md:13]: This should be a note in the startx section about how to start arbitrary programs together with your X session.

File: src/config/media/pipewire.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+If using [Sway](https://swaywm.org/), start it from the configuration file:
+
+```
+exec pipewire
+```
```
Comment (ericonr) [inline src/config/media/pipewire.md:19]: This shouldn't be here, IMO.

File: src/config/media/pipewire.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+pipewire &
+pipewire-pulse &
+```
+
+#### Example for Sway
+
+```
+exec pipewire
+exec pipewire-pulse
+```
```
Comment (ericonr) [inline src/config/media/pipewire.md:41]: This example can be inferred from above and isn't needed.

File: src/config/media/pipewire.md [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and reboot.
```
Comment (ericonr) [inline src/config/media/pipewire.md:44]: More important is probably to disable PulseAudio auto activation (I can add a section on that), because most people shouldn't be using the service already.

File: src/config/media/pipewire.md [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+
+...
+Server Name: PulseAudio (on PipeWire 0.3.18)
+...
+```
+
+##### Note on Bluetooth audio
+
+If having troubles with Bluetooth audio, make sure that `libspa-bluetooth`
+package is installed.
```
Comment (ericonr) [inline src/config/media/pipewire.md:59]: The package is required for bluetooth at all, so I don't think this is the correct language.

And title should be just `Bluetooth audio`

File: src/config/media/pipewire.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+## ALSA integration
+
+Install `alsa-pipewire`, then enable the PipeWire ALSA device and make it
+default:
+
+```
+# mkdir -p /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/50-pipewire.conf /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/99-pipewire-default.conf /etc/alsa/conf.d
+```
```
Comment (ericonr) [inline src/config/media/pipewire.md:70]: I think this section could have a link to `alsa` docs on its configuration directories, if you can find something.

File: src/config/media/pipewire.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+$ pw-jack <application>
+```
+
+Alternatively, override `libjack`:
+
+```
+# echo "/usr/lib/pipewire-0.3/jack" > /etc/ld.so.conf.d/pipewire-jack.conf
+# ldconfig
+```
```
Comment (ericonr) [inline src/config/media/pipewire.md:87]: You should mention this is glibc specific.

File: src/config/media/pipewire.md [hunk 12]
```diff
@@ -0,0 +1,87 @@
...
+
+Install `libjack-pipewire`.
+
+Use `pw-jack` to launch JACK clients manually:
+
+```
+$ pw-jack <application>
+```
+
+Alternatively, override `libjack`:
```
Comment (ericonr) [inline src/config/media/pipewire.md:82]:
```suggestion
Alternatively, override the library provided by `libjack`:
```

File: src/config/media/pipewire.md [hunk 13]
```diff
@@ -0,0 +1,87 @@
...
+# mkdir -p /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/50-pipewire.conf /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/99-pipewire-default.conf /etc/alsa/conf.d
+```
+
+## JACK replacement
+
+Install `libjack-pipewire`.
+
+Use `pw-jack` to launch JACK clients manually:
```
Comment (ericonr) [inline src/config/media/pipewire.md:76]:
```suggestion
Use [pw-jack(1)](link to manpage) to launch JACK clients manually:
```

File: src/config/media/pipewire.md [hunk 14]
```diff
@@ -0,0 +1,87 @@
...
+
+```
+exec pipewire
+exec pipewire-pulse
+```
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and reboot.
+
+To check if the replacement is working correctly, use the following command:
```
Comment (ericonr) [inline src/config/media/pipewire.md:46]:
```suggestion
To check if the replacement is working correctly, use [pactl(1)](link):
```

File: src/config/media/pipewire.md [hunk 15]
```diff
@@ -0,0 +1,87 @@
+# PipeWire
+
+To use PipeWire, install the `pipewire` package.
+
+PipeWire should be started as a user. Use the autostarting mechanism of your
+desktop environment.
```
Comment (ericonr) [inline src/config/media/pipewire.md:6]: Linking `[pipewire(1)](link)` somewhere around here would be nice too.

File: src/config/media/pipewire.md (hunk 5 above)
Comment (kkga) [reply src/config/media/pipewire.md]: Thanks, I've added a note on this in the startx section. Let me know if it looks good to you.

File: src/config/media/pipewire.md (hunk 8 above)
Comment (kkga) [reply src/config/media/pipewire.md:18]: What do you mean by disabling auto activation?
I think pulseaudio should not start automatically if the pipewire-pulse server is running.

File: src/config/media/pipewire.md (hunk 9 above)
Comment (kkga) [reply src/config/media/pipewire.md:34]: It was required for bluetooth audio to work in my case. But let me know if we should remove this.

File: src/config/media/pipewire.md (hunk 8 above)
Comment (paper42) [reply src/config/media/pipewire.md:18]: Pulseaudio can start automatically before pipewire-pulse is ran which creates a conflict between them.

File: src/config/media/pipewire.md (hunk 9 above)
Comment (paper42) [reply src/config/media/pipewire.md:34]: What @ericonr is saying is that "If having troubles" is not the correct language when libspa-bluetooth is required for bluetooth audio to work.

File: src/config/graphical-session/xorg.md [hunk 16]
```diff
@@ -41,17 +41,17 @@ Modesetting requires the `mesa-dri` package, and no additional vendor-specific
...
 For advanced configuration, see
 [modesetting(4)](https://man.voidlinux.org/modesetting.4).
 
 ### Proprietary Drivers
 
 Void also provides [proprietary NVIDIA drivers](./graphics-drivers/nvidia.md),
-which are available in the [nonfree
-repository](../../xbps/repositories/index.md#nonfree).
+which are available in the
+[nonfree repository](../../xbps/repositories/index.md#nonfree).
```
Comment (paper42) [inline src/config/graphical-session/xorg.md:54]: These changes don't seem to be related to pipewire, is it a mistake?

File: src/config/media/pipewire.md (hunk 9 above)
Comment (kkga) [reply src/config/media/pipewire.md:34]: Oh I misread that, I thought @ericonr said "is _not_ required for bluetooth at all" :)

Comment (kkga) [reply src/config/media/pipewire.md]: updated

File: src/config/graphical-session/xorg.md (hunk 16 above)
Comment (kkga) [reply src/config/graphical-session/xorg.md:54]: auto-formatting of my text editor :/ i'll revert this

File: src/config/media/index.md [hunk 17]
```diff
@@ -5,4 +5,6 @@
 To setup audio on your Void Linux system you have to decide if you want to use
 [PulseAudio](./pulseaudio.md) or just [ALSA](./alsa.md).
 
-Some applications require PulseAudio, especially closed source programs.
+Some applications require PulseAudio, especially closed source programs. As an
+alternative to PulseAudio, [PipeWire](./pipewire.md) is a new multimedia
+framework that provides its own implementation of a PulseAudio server.
```
Comment (paper42) [inline src/config/media/index.md:10]: Maybe you could be more explicit about pipewire-pulse being a drop in replacement for pulseaudio. Some people might interpret this as "If I want to run closed source applications, I need pulseaudio, pipewire will not work". Replacing the word "alternative" with "drop in replacement" could help a bit.

File: src/config/media/pipewire.md [hunk 18]
```diff
@@ -0,0 +1,63 @@
...
+`/etc/pipewire/pipewire.conf` for auto-starting the PulseAudio server, it's
+recommended to start it from the same place where you start PipeWire by
+executing `pipewire-pulse`. This keeps PipeWire's configuration file unmodified
+for smoother future upgrades.
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and reboot.
+
+To check if the replacement is working correctly, use
+[pactl(1)](https://man.voidlinux.org/pactl):
```
Comment (paper42) [inline src/config/media/pipewire.md:21]: What do you think about mentioning the package which contains pactl here? It was recently split from pulseaudio to pulseaudio-utils.

File: src/config/media/pipewire.md (hunk 3 above)
Comment (prez) [reply src/config/media/pipewire.md:55]: What is the problem exactly for musl? I remember it working effortlessly, although right now I am on glibc and cannot test.
I would prefer for the instructions to be complete with respect to musl as well.

Comment (ericonr) [reply src/config/media/pipewire.md:55]: Writing a path to musl's config file and forgetting to also include `/lib` means your system won't find any libraries. And now you can't even turn off your computer because `sudo` doesn't work.

**Comment by kkga** [comment]
@paper42 Thanks for feedback. I think I addressed all your points.
@ericonr I think the only thing that's missing is a note about disabling PulseAudio auto-activation? You mentioned you can add that.

**Comment by ericonr** [comment]
Done in c8f806f855d04bc37f1b76ee1361a1b1731911b0

File: src/config/media/pipewire.md [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+## PulseAudio replacement
+
+PipeWire can start a PulseAudio server. While it's possible to modify
+`/etc/pipewire/pipewire.conf` for auto-starting the PulseAudio server, it's
+recommended to start it from the same place where you start PipeWire by
+executing `pipewire-pulse`. This keeps PipeWire's configuration file unmodified
+for smoother future upgrades.
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and reboot.
```
Comment (ericonr) [inline src/config/media/pipewire.md:18]: Reboot sounds like the wrong advice. I think this should be worded to "make sure no other PulseAudio server instance is running".

File: src/config/media/pipewire.md [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+To check if the replacement is working correctly, use
+[pactl(1)](https://man.voidlinux.org/pactl) (provided by the `pulseaudio-utils`
+package):
+
+```
+$ pactl info
+
+...
+Server Name: PulseAudio (on PipeWire 0.3.18)
+...
```
Comment (ericonr) [inline src/config/media/pipewire.md:29]:
```suggestion
[...]
Server Name: PulseAudio (on PipeWire 0.3.18)
[...]
```

File: src/config/media/pipewire.md [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+```
+
+##### Bluetooth audio
```
Comment (ericonr) [inline src/config/media/pipewire.md:32]: I think this should be `##`?

File: src/config/media/pipewire.md [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+
+Use [pw-jack(1)](https://man.voidlinux.org/pw-jack) to launch JACK clients
+manually:
+
+```
+$ pw-jack <application>
+```
+
+Alternatively, override the library provided by `libjack`. The following
+approach will work on `glibc`:
```
Comment (ericonr) [inline src/config/media/pipewire.md:59]:
```suggestion
Alternatively, override the library provided by `libjack`. The following
approach will work on glibc-based systems:
```

File: src/config/graphical-session/xorg.md [hunk 23]
```diff
@@ -106,6 +106,16 @@ be used to start X sessions from the console. For example, to use i3, edit
 exec /bin/i3
 ```
 
+To start arbitrary programs together with an X session, add them in `~/.xinitrc`
+before the last line. For example, to start
+[pipewire(1)][https://man.voidlinux.org/pipewire] before starting i3, add the
```
Comment (ericonr) [inline src/config/graphical-session/xorg.md:111]:
```suggestion
[pipewire(1)][https://man.voidlinux.org/pipewire.1] before starting i3, add the
```

File: src/config/media/pipewire.md [hunk 24]
```diff
@@ -0,0 +1,64 @@
+# PipeWire
+
+To use PipeWire, install the `pipewire` package.
+
+[pipewire(1)](https://man.voidlinux.org/pipewire) should be started as a user.
```
Comment (ericonr) [inline src/config/media/pipewire.md:5]:
```suggestion
[pipewire(1)](https://man.voidlinux.org/pipewire.1) should be started as a user.
```

File: src/config/media/pipewire.md [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+Use the autostarting mechanism of your desktop environment or
+[startx](../graphical-session/xorg.md#startx).
+
+## PulseAudio replacement
+
+PipeWire can start a PulseAudio server. While it's possible to modify
+`/etc/pipewire/pipewire.conf` for auto-starting the PulseAudio server, it's
```
Comment (ericonr) [inline src/config/media/pipewire.md:12]:
```suggestion
`[pipewire.conf(5)](link)` for auto-starting the PulseAudio server, it's
```

File: src/config/media/pipewire.md [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+recommended to start it from the same place where you start PipeWire by
+executing `pipewire-pulse`. This keeps PipeWire's configuration file unmodified
+for smoother future upgrades.
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and reboot.
+
+To check if the replacement is working correctly, use
+[pactl(1)](https://man.voidlinux.org/pactl) (provided by the `pulseaudio-utils`
```
Comment (ericonr) [inline src/config/media/pipewire.md:21]:
```suggestion
[pactl(1)](https://man.voidlinux.org/pactl.1) (provided by the `pulseaudio-utils`
```

File: src/config/media/pipewire.md [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+# mkdir -p /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/50-pipewire.conf /etc/alsa/conf.d
+# ln -s /usr/share/alsa/alsa.conf.d/99-pipewire-default.conf /etc/alsa/conf.d
+```
+
+## JACK replacement
+
+Install `libjack-pipewire`.
+
+Use [pw-jack(1)](https://man.voidlinux.org/pw-jack) to launch JACK clients
```
Comment (ericonr) [inline src/config/media/pipewire.md:51]:
```suggestion
Use [pw-jack(1)](https://man.voidlinux.org/pw-jack.1) to launch JACK clients
```

File: src/config/media/pipewire.md (hunk 22 above)
Comment (ericonr) [inline src/config/media/pipewire.md:59]: Maybe link `ld.so(8)`?

**Review by ericonr — COMMENTED** [review]
Man page links should include the section.

File: src/config/media/pipewire.md (hunk 22 above)
Comment (ericonr) [reply src/config/media/pipewire.md]: I prefer helping at least a bit. You can still copy from it, but you can also read to try and understand what is being done :)

Comment (kkga) [reply src/config/media/pipewire.md]: Right, i've added a link.

File: src/SUMMARY.md [hunk 28]
```diff
@@ -57,6 +57,7 @@
       - [ALSA](./config/media/alsa.md)
       - [PulseAudio](./config/media/pulseaudio.md)
       - [sndio](./config/media/sndio.md)
+      - [PipeWire](./config/media/pipewire.md)
```
Comment (ericonr) [inline src/SUMMARY.md:60]: This part could be left in alphabetical order, IMO.

File: src/config/graphical-session/xorg.md [hunk 29]
```diff
@@ -106,6 +106,16 @@ be used to start X sessions from the console. For example, to use i3, edit
...
+To start arbitrary programs together with an X session, add them in `~/.xinitrc`
+before the last line. For example, to start
+[pipewire(1)](https://man.voidlinux.org/pipewire.1) before starting i3, add the
+following:
+
+```
+pipewire &
+exec /bin/i3
+```
+
```
Comment (ericonr) [inline src/config/graphical-session/xorg.md:118]: If possible, split this change into a separate commit.

File: src/config/media/pipewire.md [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+## Bluetooth audio
+
+For bluetooth audio to work, install the `libspa-bluetooth` package.
+
+## ALSA integration
+
+Install `alsa-pipewire`, then enable the PipeWire ALSA device and make it
+default:
```
Comment (ericonr) [inline src/config/media/pipewire.md:40]:
```suggestion
Install `alsa-pipewire`, then enable the PipeWire ALSA device and make it
the default:
```

**Review by ericonr — COMMENTED** [review]
Please clean up the commit message as well :)

**Comment by kkga** [comment]
> Please clean up the commit message as well :)

Sure, but can you clarify? I think the commit message is in accordance with [the contributing guide](https://github.com/void-linux/void-docs/blob/master/CONTRIBUTING.md#making-changes)?

**Comment by ericonr** [comment]
The commit title is, but the body is a bit too unnecessarily noisy:

```
Update src/config/media/pipewire.md

Co-authored-by: Érico Nogueira Rolim <34201958+ericonr@users.noreply.github.com>

Update src/config/media/pipewire.md

Co-authored-by: Érico Nogueira Rolim <34201958+ericonr@users.noreply.github.com>

pipewire.md: add a page on setting up pipewire

Update src/config/media/pipewire.md

Co-authored-by: Érico Nogueira Rolim <34201958+ericonr@users.noreply.github.com>

mention pipewire-pulse on media page

suggest starting pipewire-pulse manually instead of modifying the config

typo
```

From https://github.com/void-linux/void-docs/pull/540/commits/51b46d7aa8e390e45e728fdce378efdfa27254d0

**Comment by ericonr** [comment]
Please ping me when that's fixed, this looks good to merge! Thanks a lot for working on it, it also helped me set up `pipewire` locally :)

**Comment by kkga** [comment]
@ericonr Removed all the extra text from the commit message.

File: src/config/media/pipewire.md [hunk 31]
```diff
@@ -0,0 +1,66 @@
...
+
+PipeWire can start a PulseAudio server. While it's possible to modify
+[pipewire.conf(5)](https://man.voidlinux.org/pipewire.conf.5) for auto-starting
+the PulseAudio server, it's recommended to start it from the same place where
+you start PipeWire by executing `pipewire-pulse`. This keeps PipeWire's
+configuration file unmodified for smoother future upgrades.
+
+After setting up `pipewire-pulse`, make sure that the PulseAudio service is
+[disabled](../services/index.md#disabling-services) and make sure no other
+PulseAudio server instance is running.
```
Comment (paper42) [inline src/config/media/pipewire.md:19]: I think the word "After" is not exactly right here, people should check if the pulseaudio server is not running **before** "setting up" pipewire-pulse.

Also, "setting up" is not exactly right, I think it should be something like "starting" or "running".

File: src/config/media/pipewire.md [hunk 32]
```diff
@@ -0,0 +1,66 @@
...
+[pipewire(1)](https://man.voidlinux.org/pipewire.1) should be started as a user.
+Use the autostarting mechanism of your desktop environment or
+[startx](../graphical-session/xorg.md#startx).
+
+## PulseAudio replacement
+
+PipeWire can start a PulseAudio server. While it's possible to modify
+[pipewire.conf(5)](https://man.voidlinux.org/pipewire.conf.5) for auto-starting
+the PulseAudio server, it's recommended to start it from the same place where
+you start PipeWire by executing `pipewire-pulse`. This keeps PipeWire's
```
Comment (paper42) [inline src/config/media/pipewire.md:14]: "PipeWire's configuration file" sounds a bit weird to me, what about "the PipeWire configuration file"? Keep in mind that I am not a native speaker, so I might be wrong.

File: src/config/graphical-session/xorg.md [hunk 33]
```diff
@@ -106,6 +106,16 @@ be used to start X sessions from the console. For example, to use i3, edit
 exec /bin/i3
 ```
 
+To start arbitrary programs together with an X session, add them in `~/.xinitrc`
+before the last line. For example, to start
```
Comment (Piraty) [inline src/config/graphical-session/xorg.md:110]: ` <...> before starting the window manager / DE.`
Makes the example code block is superfluous in my opinion (or could be integrated in the above code block even)

File: src/config/media/index.md [hunk 34]
```diff
@@ -3,6 +3,8 @@
 ## Audio setup
 
 To setup audio on your Void Linux system you have to decide if you want to use
-[PulseAudio](./pulseaudio.md) or just [ALSA](./alsa.md).
+[PulseAudio](./pulseaudio.md), [PipeWire](./pipewire.md) or just
+[ALSA](./alsa.md).
 
-Some applications require PulseAudio, especially closed source programs.
+Some applications require PulseAudio, especially closed source programs, but
+[PipeWire](./pipewire.md) provides a drop-in PulseAudio replacement.
```
Comment (Piraty) [inline src/config/media/index.md:10]: `<..> drop-in replacement for PulseAudio.`

**Review by Piraty — CHANGES_REQUESTED** [review]

File: src/config/media/pipewire.md (hunk 31 above)
Comment (kkga) [reply src/config/media/pipewire.md:19]: Sounds good

File: src/config/media/pipewire.md (hunk 32 above)
Comment (kkga) [reply src/config/media/pipewire.md:14]: Correct, I'll update this

File: src/config/graphical-session/xorg.md (hunk 33 above)
Comment (kkga) [reply src/config/graphical-session/xorg.md:107]: True. I've tweaked this a bit.

File: src/config/media/index.md (hunk 34 above)
Comment (kkga) [reply src/config/media/index.md]: :+1:

File: src/config/graphical-session/xorg.md [hunk 35]
```diff
@@ -99,14 +99,23 @@ If there is a match, modesetting is being used.
...
+be used to start X sessions from the console. For example, to start
+[i3(1)](https://man.voidlinux.org/i3.1), edit `~/.xinitrc` to contain `exec
+/bin/i3` on the last line.
+
+To start arbitrary programs together with an X session, add them in `~/.xinitrc`
+before the last line. For example, to start
+[pipewire(1)](https://man.voidlinux.org/pipewire.1) before starting i3, add
+`pipewire &` before the last line.
+
+Here is the full example of `~/.xinitrc` which starts `pipewire` and `i3`:
```
Comment (ericonr) [inline src/config/graphical-session/xorg.md:111]:
```suggestion
A `~/.xinitrc` file which starts `pipewire` and `i3` is shown below:
```
